runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaGroupClientFactory.java

```diff
@@ -0,0 +1,2515 @@
+
+        KafkaGroupStream(
+            MessageConsumer application,
+            long originId,
+            long routedId,
+            long initialId,
+            long affinity,
+            long resolvedId,
+            String16FW groupId,
+            String16FW protocol,
+            int timeout,
+            KafkaSaslConfig sasl)
+        {
+            this.application = application;
+            this.originId = originId;
+            this.routedId = routedId;
+            this.initialId = initialId;
+            this.replyId = supplyReplyId.applyAsLong(initialId);
+            this.affinity = affinity;
+            this.groupId = groupId;
+            this.protocol = protocol;
+            this.timeout = timeout;
+            this.clusterClient = new ClusterClient(routedId, resolvedId, sasl, this);
+            this.coordinatorClient = new CoordinatorClient(routedId, resolvedId, sasl, this);
+        }
+
+        private void onApplication(
+            int msgTypeId,
+            DirectBuffer buffer,
+            int index,
+            int length)
+        {
+            switch (msgTypeId)
+            {
+            case BeginFW.TYPE_ID:
+                final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+                onApplicationBegin(begin);
+                break;
+            case DataFW.TYPE_ID:
+                final DataFW data = dataRO.wrap(buffer, index, index + length);
+                onApplicationData(data);
+                break;
+            case EndFW.TYPE_ID:
+                final EndFW end = endRO.wrap(buffer, index, index + length);
+                onApplicationEnd(end);
+                break;
+            case AbortFW.TYPE_ID:
+                final AbortFW abort = abortRO.wrap(buffer, index, index + length);
+                onApplicationAbort(abort);
+                break;
+            case WindowFW.TYPE_ID:
+                final WindowFW window = windowRO.wrap(buffer, index, index + length);
+                onApplicationWindow(window);
+                break;
+            case ResetFW.TYPE_ID:
+                final ResetFW reset = resetRO.wrap(buffer, index, index + length);
+                onApplicationReset(reset);
+                break;
+            default:
+                break;
+            }
+        }
+
+        private void onApplicationBegin(
+            BeginFW begin)
+        {
+            final long traceId = begin.traceId();
+            final long authorization = begin.authorization();
+
+            state = KafkaState.openingInitial(state);
+
+            clusterClient.doNetworkBegin(traceId, authorization, affinity);
+            doApplicationWindow(traceId, 0L, 0, 0, 0);
+        }
+
+        private void onApplicationData(
+            DataFW data)
+        {
+            final long traceId = data.traceId();
+            final long budgetId = data.budgetId();
+
+            coordinatorClient.doSyncRequest(traceId, budgetId, data.payload());
```

Comment:
This is what I am not sure.

Reply:
I think this is the case where the leader is making assignments to the members, right?

First, we need to verify and reject if not the leader, as Kafka will reject it anyway.
Then we need to `doSyncRequest` to update the assignments, but the leader is already in the group, so no need to proactively re-Join the group, i think, agree?